pkg/util/admissioncheck/admissioncheck.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admissioncheck
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+var (
+	ErrNilParametersRef = errors.New("nil parameters reference")
+	ErrBadParametersRef = errors.New("bad parameters reference")
+)
+
+type ConfigHelper[PCT interface {
```

Comment:
Why are we generalizing the ConfigHelper? Don't do it if not needed, if needed it should be reflected in the title and description of the PR.

Reply:
Oh I see, the helper is generalized in this WIP PR (https://github.com/kubernetes-sigs/kueue/pull/1313/files) to support the new `DelegateConfig` type. However, it is not clear for me we need this type. Is it KEPed / discussed somewhere we could reference? 

Actually, IIUC, the currently review-pending KEP assumes the configurations are synchronized (copied) by the user, so maybe there is no reason to introduce the delegating config? 

If not, I suggest to scope this PR just to the package change (which is good on its own), because AdmissionChecks are not just for provisioning.

Reply:
#1313 and  #1380 will eventually get in sync

Reply:
I see, I'm ok with proceeding with the generic Config, because it is no API change, even though I'm not 100% sure it will be needed.